vsphere/data_source_vsphere_datastore_cluster.go

```diff
@@ -24,15 +26,39 @@ func dataSourceVSphereDatastoreCluster() *schema.Resource {
 				Optional:    true,
 				Description: "The managed object ID of the datacenter the cluster is located in. Not required if using an absolute path.",
 			},
+			"datastores": {
+				Type:        schema.TypeSet,
+				Computed:    true,
+				Description: "The names of datastores in the datastorecluster.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
 		},
 	}
 }
 
 func dataSourceVSphereDatastoreClusterRead(d *schema.ResourceData, meta interface{}) error {
+	ctx := context.Background()
+	client := meta.(*Client).vimClient
 	pod, err := resourceVSphereDatastoreClusterGetPodFromPath(meta, d.Get("name").(string), d.Get("datacenter_id").(string))
 	if err != nil {
 		return fmt.Errorf("error loading datastore cluster: %s", err)
 	}
 	d.SetId(pod.Reference().Value)
+	dsNames := []string{}
+	podDatastores, err := pod.Children(ctx)
+	if err != nil {
+		return fmt.Errorf("error retrieving datastores in datastore cluster: %s", err)
+	}
+	for d := range podDatastores {
+		ds, err := datastore.FromID(client, podDatastores[d].Reference().Value)
+		if err != nil {
+			return fmt.Errorf("error retrieving datastore: %s", err)
+		}
+		dsNames = append(dsNames, ds.Name())
+	}
```

Comment:
```suggestion
	childDatastores, err := pod.Children(ctx)
	if err != nil {
		return fmt.Errorf("error retrieving datastores in datastore cluster: %s", err)
	}
	for d := range childDatastores {
		ds, err := datastore.FromID(client, childDatastores[d].Reference().Value)
		if err != nil {
			return fmt.Errorf("error retrieving datastore: %s", err)
		}
		dsNames = append(dsNames, ds.Name())
	}
```

It would make sense for this to be a bit more more similar to the changes seen in https://github.com/hashicorp/terraform-provider-vsphere/pull/2273.

Reply:
Applied the changes. Let me know if the current is similar enough so I can resolve this thread.